mesh/mesh.go

```diff
@@ -205,54 +190,54 @@ func (m *Mesh) handleOrphanBlocks(block *Block) {
 				break
 			}
 		}
-
 	}
 }
 
-func (m *Mesh) GetOrphanBlocksByLayerId(layerId LayerID) []BlockID {
-	m.orphMutex.RLock()
-	defer m.orphMutex.RUnlock()
-	if keys, has := m.orphanBlocks[layerId]; has {
-		ids := make([]BlockID, 0, len(keys))
-
-		for bid := range keys {
-			ids = append(ids, bid)
-		}
-		return ids
-
+func (m *Mesh) GetUnverifiedLayerBlocks(l LayerID) []BlockID {
+	x, err := m.meshDB.layers.Get(l.ToBytes())
+	if err != nil {
+		panic(fmt.Sprintf("could not retrive latest layer = %d blocks ", l))
 	}
-	return make([]BlockID, 0, 0)
-}
-
-//todo better thread safety
-func (m *Mesh) GetOrphanBlocks() []BlockID {
-	m.orphMutex.RLock()
-	defer m.orphMutex.RUnlock()
-	keys := make([]BlockID, 0, m.orphanBlockCount)
-	for _, val := range m.orphanBlocks {
-		for bid := range val {
-			keys = append(keys, bid)
-		}
+	blockIds, err := bytesToBlockIds(x)
+	if err != nil {
+		panic(fmt.Sprintf("could bytes to id array for layer %d ", l))
 	}
-
-	return keys
+	arr := make([]BlockID, 0, len(blockIds))
+	for bid := range blockIds {
+		arr = append(arr, bid)
+	}
+	return arr
 }
 
 //todo better thread safety
 func (m *Mesh) GetOrphanBlocksExcept(l LayerID) []BlockID {
 	m.orphMutex.RLock()
 	defer m.orphMutex.RUnlock()
-	keys := make([]BlockID, 0, m.orphanBlockCount)
+	ids := map[BlockID]struct{}{}
 	for key, val := range m.orphanBlocks {
-		if key == l {
-			continue
-		}
-		for bid := range val {
-			keys = append(keys, bid)
+		if key < l {
+			for bid := range val {
+				ids[bid] = struct{}{}
+			}
 		}
 	}
 
-	return keys
+	prevLayer, err := m.getLayer(l - 1)
+	if err != nil {
+		panic(fmt.Sprint("failed getting latest layer  ", err))
+	}
+
+	//add last layer blocks
+	for _, b := range prevLayer.Blocks() {
+		ids[b.ID()] = struct{}{}
+	}
+
+	idArr := make([]BlockID, 0, len(ids))
+	for i := range ids {
```

Comment:
we cant use a map to create this slice, the blocks must be in the same order in all nodes. iterating a map is random and therefore will not produce same results